micro-rdk/src/common/conn/server.rs

```diff
@@ -296,40 +325,81 @@ where
         app_connector: C,
         app_config: AppClientConfig,
         max_concurent_connections: usize,
+        app_client_tasks: Vec<Box<dyn PeriodicAppClientTask>>,
     ) -> Self {
         Self {
             http_listener,
             webrtc_config,
             exec,
             app_connector,
             app_config,
-            app_client: None,
+            app_client: Default::default(),
             incoming_connection_manager: IncomingConnectionManager::new(max_concurent_connections),
+            app_client_tasks,
         }
     }
     pub async fn serve(&mut self, robot: Arc<Mutex<LocalRobot>>) {
         let cloned_robot = robot.clone();
+
```

Comment:
Transforms each element of the vector of boxed PeriodicAppClientTask implementors into an actual async task, and collects them into a vector so we don't need to detach.